plugins/mongodb/README.md

```diff
@@ -0,0 +1,202 @@
+# MongoDB plugin for ECL
+
+The MongoDB plugin allows an ECL user to embed MongoDB function calls into their code and run it
+on the HPCC Platform. The plugin supports inserting a dataset into a database using `insert_many`, and can 
+build ECL datasets from MongoDB result documents from the `find`, `update`, `delete`, `aggregate`, and `runCommand` methods.
+
+The embedded script that gets passed to the plugin can be used to create complex documents to support almost every
+MongoDB command. Parameters of the function definition can be inserted into the document using the '$' character. This is useful if you
+want to have a document contain a numerical type since every other value in the embedded script gets bound to a string.
+
+It is important to use the same keys as the ones in MongoDB when declaring a return type or when creating a BSON document. Otherwise, the plugin will look for a field that might not exist to return when building the resulting dataset.
+
+## Installation
+
+The plugin uses vcpkg and can be installed by creating a seperate build directory from the platform and running the following commands:
+```
+cd ./mongodb-build
+cmake -DMONGODBEMBED=ON ../HPCC-Platform
+make -j3 package
+sudo dpkg -i ./hpccsystems-plugin-mongodbembed_<version>.deb
+```
+
+## Documentation
+
+[Doxygen](https://www.doxygen.nl/index.html) can be used to create nice HTML documentation for the code. Call/caller graphs are also generated for functions if you have [dot](https://www.graphviz.org/download/) installed and available on your path.
+
+Assuming `doxygen` is on your path, you can build the documentation via:
+```
+cd plugins/mongodb
+doxygen Doxyfile
+```
+
+The Documentation can then be accessed via `mongodb/docs/html/index.html`.
+
+## Usage
+
+To start using the plugin you must import mongodb before you use it in the EMBED statement.
+```
+Import MongoDB;
+```
+
+You then need to declare a function that embeds the MongoDB plugin. Within the EMBED statement is where you can write code to interact with the MongoDB database or collection that you have passed into the EMBED statement. The plugin only supports one line and one operation per EMBED statement. If you wish to chain multiple operations together the aggregate function is supported and can take an aggregation pipeline as long as it is valid BSON.
+
+To create the uri for the MongoDB connection instance the ECL user needs to pass in the username, password, server name, or just the port to use for connecting to the cluster. The plugin takes all of these and creates a shared connection instance for many threads to have access to the MongoDB databases. The plugin can only connect to one MongoDB cluster per workunit, but it can access multiple databases and collections if multiple functions are defined. The ECL user can define the batch size of the result rows.
```

Comment:
Suggestion:  Add a list of extra arguments (user, password, server, etc) and what they mean, along with example values if appropriate.